smoke-tests/run_smoke_test.sh

```diff
@@ -86,19 +86,37 @@ has_intercept_id() {
 }
 
 # Puts preview url in a variable
-# Not currently used bc curl doesn't work with preview url bc we need
-# cookie system A adds (I think)
 get_preview_url() {
-    local regex="Preview URL : (https://[^ >]+)"
-    if [[ $output =~ $regex ]]; then
-        # shellcheck disable=SC2034
-        previewurl="${BASH_REMATCH[1]}"
-    else
+    preview_url=$(echo "$output" | grep -Eo 'https://[^ >]+')
+    if [[ -z $preview_url ]]; then
         echo "No Preview URL found"
         exit 1
     fi
 }
 
+# Puts workstation api key in a variable
+get_workstation_apikey() {
+    local cache_file
+    uname=$(uname)
+    if [ "$uname" == "Darwin" ]; then
+        cache_file="$HOME/Library/Caches/telepresence/apikeys.json"
+    elif [ "$uname" == "Linux" ]; then
+        if [ -n "$XDG_CONFIG_HOME" ]; then
+            cache_file="$XDG_CONFIG_HOME/telepresence/apikeys.json"
+        else
+            cache_file="$HOME/.cache/telepresence/apikeys.json"
+        fi
+    fi
```

Comment:
A pair of style suggestions that you are free to ignore:
- I'd have written this as a switch/case
- use `:-` for fallbacks for unset variables: `${VARNAME:-DEFAULT_VALUE}`
```bash
case "$(uname)" in
	Darwin)
		cache_file="$HOME/Library/Caches/telepresence/apikeys.json"
		;;
	Linux)
		cache_file="${XDG_CONFIG_HOME:-$HOME/.cache}/telepresence/apikeys.json"
		;;
esac
```

Reply:
Ah yeah I should have looked up switch statements in bash, since this is an obvious candidate since the # of OS' is likely to grow (i.e. windows is imminent and we should probably make sure this can run on windows too.. will add that to my list of cooldown activities)